<!-- gate.launch -->
<launch>
  <!-- ====== 파라미터 (필요하면 여기서 숫자 바꿔서 쓸 것) ====== -->
  <arg name="eps"              default="0.4" />
  <arg name="min_samples"      default="10" />
  <arg name="basic_speed"      default="1000.0" />
  <arg name="stop_speed"       default="0.0" />
  <arg name="gate_down_thresh" default="3" />
  <arg name="gate_up_thresh"   default="3" />
  <arg name="gate_stop_dist"   default="3.0" />

  <node pkg="jiwon_mission"
        type="gate"
        name="mission7_gate_node"
        output="screen">

    <!-- ====== gate_node private params (~) ====== -->
    <param name="eps"              value="$(arg eps)" />
    <param name="min_samples"      value="$(arg min_samples)" />
    <param name="basic_speed"      value="$(arg basic_speed)" />
    <param name="stop_speed"       value="$(arg stop_speed)" />
    <param name="gate_down_thresh" value="$(arg gate_down_thresh)" />
    <param name="gate_up_thresh"   value="$(arg gate_up_thresh)" />
    <param name="gate_stop_dist"   value="$(arg gate_stop_dist)" />

    <!-- 필요하면 토픽 remap
    <remap from="/scan" to="/scan_filtered"/>
    <remap from="/commands/motor/speed" to="/my_car/speed_cmd"/>
    -->
  </node>

  <!-- RViz에서 /gate_cluster_marker 보고 싶으면 rviz 같이 띄우기 (옵션) -->
  
  <node pkg="rviz" type="rviz" name="rviz" output="screen"
        args="-d $(find jiwon_mission)/rviz/mission7_gate.rviz"/>
</launch>
